### PR TITLE
Fix limit check to Add/Remove liquidity in ERC4626 pools

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

When an ERC4626 pool does not have all tokens ERC4626 and a user adds/removes liquidity in the pool, the limit of the token that is not wrapped is not being checked. This PR fixes it.

## Type of change

- [x] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/vault/test/foundry/CompositeLiquidityRouterERC4626Pool.t.sol [hunk 1]
```diff
@@ -434,6 +434,44 @@ contract CompositeLiquidityRouterERC4626PoolTest is BaseERC4626BufferTest {
         );
     }
 
+    function testAddLiquidityProportionalToPartialERC4626PoolAboveLimit() public {
+        // Make sure the operation is within the buffer liquidity.
+        uint256 operationAmount = bufferInitialAmount;
+
+        uint256[] memory maxAmountsIn = new uint256[](2);
+        maxAmountsIn[partialWaDaiIdx] = operationAmount;
+        maxAmountsIn[partialUsdcIdx] = 1;
```
Comment (jubeira) [inline pkg/vault/test/foundry/CompositeLiquidityRouterERC4626Pool.t.sol:443]: Could set this to expected - 1 to test the very edge right?

File: pkg/vault/test/foundry/CompositeLiquidityRouterERC4626Pool.t.sol [hunk 2]
```diff
@@ -624,6 +662,41 @@ contract CompositeLiquidityRouterERC4626PoolTest is BaseERC4626BufferTest {
...
+        uint256[] memory expectedWrappedAmountsOut = router.queryRemoveLiquidityProportional(
+            partialErc4626Pool,
+            exactBptAmountIn,
+            address(this),
+            bytes("")
+        );
+        vm.revertTo(snapshot);
+
+        uint256[] memory minAmountsOut = new uint256[](2);
+        minAmountsOut[partialUsdcIdx] = 10 * expectedWrappedAmountsOut[partialUsdcIdx];
```
Comment (jubeira) [inline pkg/vault/test/foundry/CompositeLiquidityRouterERC4626Pool.t.sol:679]: Why the 10x margin?

**Review by jubeira — APPROVED** [review]
Just a few comments  / questions; otherwise LGTM!

**Review by elshan-eth — APPROVED** [review]
I agree with @jubeira comments but otherwise LGTM

**Pull request merged**